app/src/main/java/camp/nextstep/DispatcherServlet.java

```diff
@@ -30,14 +42,21 @@ protected void service(final HttpServletRequest request, final HttpServletRespon
         log.debug("Method : {}, Request URI : {}", request.getMethod(), requestURI);
 
         try {
-            final var controller = manualHandlerMapping.getHandler(requestURI);
-            final var viewName = controller.execute(request, response);
-            var jspView = new JspView(viewName);
-            jspView.render(null, request, response);
+            execute(request, response);
         } catch (Throwable e) {
             log.error("Exception : {}", e.getMessage(), e);
             throw new ServletException(e.getMessage());
         }
     }
 
+    private void execute(HttpServletRequest request, HttpServletResponse response) throws Exception {
+
+        Optional<HandlerMapping> handler = handlers.stream()
+                .filter(it -> it.supports(request))
+                .findFirst();
+
+        if (handler.isPresent()) {
+            handler.get().adapt(handler.get().getHandler(request), request, response);
+        }
+    }
```

Comment:
```suggestion
    private void execute(HttpServletRequest request, HttpServletResponse response) throws Exception {

        Optional<HandlerMapping> handler = handlers.stream()
                .filter(it -> it.supports(request))
                .findFirst();

        handler.ifPresent(currying(request, response));
    }

    private Consumer<HandlerMapping> currying(HttpServletRequest request, HttpServletResponse response) {
        return handler -> {
            try {
                handler.adapt(handler.getHandler(request), request, response);
            } catch (Exception e) {
                throw new RuntimeException(e);
            }
        };
    }
```

Optional 의 isPresent 를 조건문에 활용하는 것은 null 체크를 통한 절차지향적 코딩과 크게 다를바가 없습니다.
참고로 저는 절차지향 무척 좋아합니다.
변경을 요구하기 보다는 함수형에 욕심을 내면 이런식의 표현도 가능하다는 느낌으로 예시코드를 작성해보았습니다.
혹시 함수형에 욕심있으시다면, 람다에서의 try catch 부분을 깔끔하게 만드는 데에 도전해보시면 재미있습니다! ㅋㅋ

도전하시지 않더라도 currying 이라고 하는 개념에 대해서 공부해보시면 재밌습니다.

Reply:
아하! `쏙쏙 들어오는 함수형 코딩`이라는 책에서 `함수 본문을 콜백으로 바꾸기` 라는 내용이 있는데 비슷한거같네요!

`handler.ifPresent(currying(request, response));` 이렇게 하니 if 분기문 보다 코드 가독성도 훨씬 올라가네요!